### PR TITLE
feat: Organization + Membership

### PR DESCRIPTION
**Pull request by kanatsanan6**
Todos-app: Single-tenancy -> Muti-tenancy

- [x] Add Organization
- [x] Add Membership (user can be a member of organizations)
- [x] UI for Organization / Membership
Screen Shot 2565-08-08 at 13 54 22

Screen Shot 2565-08-08 at 13 53 40

### CODE REVIEW

File: app/controllers/companies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  def set_company
+    @company = Company.find(params[:id])
+  end
+
+  def company_params
+    params.require(:company).permit(:name).merge(user: current_user)
+  end
+
+  def check_user
+    redirect_to companies_path and return unless @company.user_id == current_user.id
```
Comment (Supermalonz) [inline app/controllers/companies_controller.rb:50]: ` redirect_to companies_path unless @company.user_id == curreny_user.id`

Comment (kanatsanan6) [reply app/controllers/companies_controller.rb:50]: will fix in https://github.com/kanatsanan6/todos-app-on-rails/pull/14 naa Tle

File: app/controllers/companies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  def update
+    if @company.update(company_params)
+      redirect_to companies_path
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    redirect_to companies_path, status: :see_other if @company.destroy
```
Comment (jsice) [inline app/controllers/companies_controller.rb:36]: what if *#destroy* returns false?

File: app/controllers/memberships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @membership = @company.memberships.find(params[:id])
+  end
+
+  def membership_params
+    @email = params[:membership][:email]
+    params.permit(:membership).merge({ user: User.find_by(email: @email) })
+  end
+
+  def order(membership)
+    membership.order(:created_at)
```
Comment (jsice) [inline app/controllers/memberships_controller.rb:44]: you should explicitly state that it is sorted in asc or desc order
```
    membership.order(created_at: :asc)
```

File: app/controllers/memberships_controller.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  def new
+    @membership = @company.memberships.new
+  end
+
+  def create
+    @membership = @company.memberships.create!(membership_params)
+    redirect_to company_memberships_path
+  rescue ActiveRecord::StatementInvalid
+    render :new, status: :unprocessable_entity
```
Comment (jsice) [inline app/controllers/memberships_controller.rb:21]: why don't use the same style as companies controller 
```
if xxx.create
  ...
else
  ...
end
```

File: app/controllers/memberships_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def destroy
+    redirect_to company_memberships_path, status: :see_other if @membership.destroy
```
Comment (jsice) [inline app/controllers/memberships_controller.rb:25]: same as above and why see other? 🤔

File: app/controllers/memberships_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    @company = Company.find(params[:company_id])
+  end
+
+  def set_membership
+    @membership = @company.memberships.find(params[:id])
+  end
+
+  def membership_params
+    @email = params[:membership][:email]
+    params.permit(:membership).merge({ user: User.find_by(email: @email) })
```
Comment (jsice) [inline app/controllers/memberships_controller.rb:40]:
```suggestion
    params.permit(:membership).merge(user: User.find_by(email: @email))
```

File: app/views/companies/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (jsice) [inline app/views/companies/edit.html.erb:1]: empty line

File: app/controllers/memberships_controller.rb (hunk 3 above)
Comment (kanatsanan6) [reply app/controllers/memberships_controller.rb:44]: fixed in https://github.com/kanatsanan6/todos-app-on-rails/pull/15/

File: app/controllers/memberships_controller.rb (hunk 4 above)
Comment (kanatsanan6) [reply app/controllers/memberships_controller.rb:21]: fixed in https://github.com/kanatsanan6/todos-app-on-rails/pull/15/

File: app/controllers/memberships_controller.rb (hunk 6 above)
Comment (kanatsanan6) [reply app/controllers/memberships_controller.rb:40]: fixed in https://github.com/kanatsanan6/todos-app-on-rails/pull/15/

File: app/views/companies/edit.html.erb (hunk 7 above)
Comment (kanatsanan6) [reply app/views/companies/edit.html.erb:1]: fixed in https://github.com/kanatsanan6/todos-app-on-rails/pull/15/

**Pull request closed without merging**